cloud/aws/providerconfig/v1alpha1/types.go

```diff
@@ -178,17 +178,28 @@ type VPC struct {
 type Subnet struct {
 	ID string `json:"id"`
 
-	VpcID            string `json:"vpcId"`
-	AvailabilityZone string `json:"availabilityZone"`
-	CidrBlock        string `json:"cidrBlock"`
-	IsPublic         bool   `json:"public"`
+	VpcID            string  `json:"vpcId"`
+	AvailabilityZone string  `json:"availabilityZone"`
+	CidrBlock        string  `json:"cidrBlock"`
+	IsPublic         bool    `json:"public"`
+	RouteTableID     *string `json:"routeTableId"`
+	NatGatewayID     *string `json:"natGatewayId"`
 }
 
 // Subnets is a slice of Subnet.
 type Subnets []*Subnet
 
+// ToMap returns a map from id to subnet.
+func (s Subnets) ToMap() (res map[string]*Subnet) {
```

Comment:
I might be missing something, since I'm still relatively new to Go, but wouldn't this cause issues due to the default value for a map being nil?

Reply:
Thanks for catching that! That's totally right